src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -307,38 +227,21 @@ private void addCredentials(Request.Builder builder) {
 
     private <T> BitbucketResponse<T> makeGetRequest(
             @Nonnull HttpUrl url, @Nonnull ObjectReader<T> reader) {
-        Request.Builder requestBuilder = new Request.Builder().url(url);
-        addCredentials(requestBuilder);
+        return httpRequestExecutor.executeGet(url, credentials,
+                    response -> new BitbucketResponse<>(
+                            response.headers().toMultimap(), unmarshall(reader, response.body())));
```

Comment:
`response.body()` can be `null`

Reply:
I don't think we need null checks everything. If the body is null, NoContentException will be thrown. That's part of the contract.

Reply:
Good point, I'd missed that